samtranslator/model/apigateway.py

```diff
@@ -93,6 +95,51 @@ def make_auto_deployable(self, stage, swagger=None):
         stage.update_deployment_ref(self.logical_id)
 
 
+class ApiGatewayResponse(object):
+    ResponseParameterProperties = ["Headers", "Paths", "QueryStrings"]
+
+    def __init__(self, api_logical_id=None, response_parameters=None, response_templates=None, status_code=None):
+        if response_parameters:
+            for response_parameter_key in response_parameters.keys():
+                if response_parameter_key not in ApiGatewayResponse.ResponseParameterProperties:
+                    raise InvalidResourceException(
+                        api_logical_id,
+                        "Invalid gateway response parameter '{}'".format(response_parameter_key))
+
+        # status_code must look like a status code, if present. Let's not be judgmental; just check 0-999.
+        if status_code and not match(r'^[0-9]{1,3}$', str(status_code)):
```

Comment:
Yes, this regex is sufficient, however, I don't think we should forbid strings here. The docs say StatusCode should be a string https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-gatewayresponse.html#cfn-apigateway-gatewayresponse-statuscode but it may accept both String and Number (with the latter getting parsed to the former)

Reply:
Sure thing. That suggests the following tests to me:

- `StatusCode: 401` => PASS
- `StatusCode: '401'` => PASS
- `StatusCode: four-oh-one` => FAIL

> [...] I don't think we should forbid strings here.

I don't think I am anymore, am I? I removed the numeric type test. I'm inexperienced at Python, but my understanding is that `str` will turn both `401` and `'401'` into `'401'`.

Reply:
Oh, I think I see now. You must mean "Property 'StatusCode' must be numeric", right?

I guess I was thinking of "numeric" as "_looks like_ a number". Do you have a suggestion of better wording that would explain allowing `200` and `'200'`?

Reply:
Ignore me, I missed the part where you were casting it to a string. I think the error message as-is is fine.

Reply:
>Ignore me [...]

Never!

>I think the error message as-is is fine.

Cool. I think I'm caught up, then.